docs/src/pages/components/data-grid/rows/VariableRowHeightGrid.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import {
+  DataGrid,
+  GridToolbarContainer,
+  GridToolbarDensitySelector,
+} from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+function CustomToolbar() {
+  return (
+    <GridToolbarContainer>
+      <GridToolbarDensitySelector />
+    </GridToolbarContainer>
+  );
+}
+
+export default function VariableRowHeightGrid() {
+  const isRowIndexOdd = React.useRef(false);
+  const { data } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: 100,
+    maxColumns: 6,
+  });
+
+  return (
+    <div style={{ height: 400, width: '100%' }}>
+      <DataGrid
+        {...data}
+        getRowHeight={() => {
+          isRowIndexOdd.current = !isRowIndexOdd.current;
```

Comment:
This logic may be a little weird for users at 1st.
If we could find another row height criteria than the odd / even index based on a mutable ref.

Non blocking

Reply:
If this is too weird I can just make a static example with custom rows and check their id?

Reply:
I think it would make the example easier to read.